### DISCUSSION TITLE
Is file-validator examining the content of a file to ensure file integrity?

### DISCUSSION THREAD

**Discussion by invidu** [file-validator/file-validator#219] [Q&A] [answered]
I'm not quite sure about the capability of file-validator. Is it capable of completely inspecting a given file to find out if a file is, say a valid jpg or mp3 or if it's corrupted? I have tinkered a bit with `ffmpeg` for that purpose, but plan to extend file integrity checking to more file types. The issue I have is that I don't have any existing file hashes to compare my files to, so I need to inspect a bunch of files to see if they're corrupt or not.

Merely looking at the magic number wouldn't work, because a corruption could have left the magic number untouched, but not the actual file data.

To that end, can you provide a small example script that would allow me to use file-validator?

-Thanks

**Comment by rzashakeri** [comment, marked as answer]
> I'm not quite sure about the capability of file-validator. Is it capable of completely inspecting a given file to find out if a file is, say a valid jpg or mp3 or if it's corrupted? I have tinkered a bit with `ffmpeg` for that purpose, but plan to extend file integrity checking to more file types. The issue I have is that I don't have any existing file hashes to compare my files to, so I need to inspect a bunch of files to see if they're corrupt or not.
> 
> Merely looking at the magic number wouldn't work, because a corruption could have left the magic number untouched, but not the actual file data.
> 
> To that end, can you provide a small example script that would allow me to use file-validator?
> 
> -Thanks

hi
yes can
What do you mean by being destructive? If you say files that are stained with the virus, it can't
If so you should use an antivirus
file validator is library for validating files Not file checking in terms of being destructive
This library is based on our mime and extension and validation size

**Reply by invidu** [reply]
Hi,
Some of my files got corrupted through a disk copy error. Unfortunately that happened to both the source and the destination disk. It's not a corruption from a virus.

Do you have a small example script that I could use to examine the use of file-validator?

-Thanks.

**Comment by rzashakeri** [comment]
> Hi, Some of my files got corrupted through a disk copy error. Unfortunately that happened to both the source and the destination disk. It's not a corruption from a virus.
> 
> Do you have a small example script that I could use to examine the use of file-validator?
> 
> -Thanks.

HI

read here https://file-validator.github.io/docs/usage/file_validator#how-used

**Reply by invidu** [reply]
Thanks, I saw that, but somehow I couldn't get a clue how I'd use that in my script. I'm (admittedly) still a newby to development.
